### PR TITLE
throw ServletException, add ImmutableList, asynchronously load photos, style maps

### PR DESCRIPTION
**Pull request by danyag0820**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/icebergDataServlet.java [hunk 1]
```diff
@@ -1,41 +1,46 @@
...
 public class icebergDataServlet extends HttpServlet {
 
+  private Collection<Iceberg> icebergSightingsArray;
   private Collection<Iceberg> icebergSightings;
 
   @Override
-  public void init() {
-    icebergSightings = new ArrayList<>();
+  public void init() throws ServletException {
+    icebergSightingsArray = new ArrayList<>();
```
Comment (saur-g) [inline portfolio/src/main/java/com/google/sps/servlets/icebergDataServlet.java:27]: You could avoid using this temporary array by using the builder,

see Example#2 on this page: https://www.programcreek.com/java-api-examples/index.php?api=com.google.common.collect.ImmutableList.Builder

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -114,3 +113,18 @@ function createIcebergSightingsMap() {
...
+    '/images/friendImages/snowman.JPG','/images/natureImages/IMG-4799.JPG','/images/friendImages/IMG-2559.JPG','/images/natureImages/IMG-4901.JPG',
+    '/images/natureImages/IMG-5025.JPG','/images/friendImages/IMG-7364.JPG','/images/natureImages/IMG-6283.JPG',
+    '/images/friendImages/IMG-8269.jpg','/images/natureImages/IMG-7379.JPG','/images/natureImages/IMG-7383.JPG',
+    '/images/friendImages/IMG-1539.JPG','/images/natureImages/IMG-7621.JPG','/images/natureImages/IMG-9385.jpg']
+    for(i = 0; i < images.length; i++) {
+        var img = new Image();
+        img.onload = document.getElementById('photos').appendChild(img);
+        img.src = images[i];
+    }
+}
```
Comment (saur-g) [inline portfolio/src/main/webapp/script.js:129]: Wow, this is impressive how you have identified the problem, researched various solutions, and finally implemented :)

Comment (danyag0820) [reply portfolio/src/main/webapp/script.js:129]: thank you :D

File: portfolio/src/main/java/com/google/sps/servlets/icebergDataServlet.java (hunk 1 above)
Comment (danyag0820) [reply portfolio/src/main/java/com/google/sps/servlets/icebergDataServlet.java:27]: sounds good!

**Pull request merged**